game/screen_songs.cc

```diff
@@ -379,17 +380,23 @@ std::string ScreenSongs::getHighScoreText() const {
 	auto stream = std::stringstream();
 	auto n = 0;
 	for (auto const& [track, scores]: scoresByTrack) {
+		if (n + 2 >= maxLines) break; // don't print a track if no scores for it would be printed
 		stream << track << ":\n";
+		n++;
```

Comment:
Use the prefix notation where possible. Should be the default. The postfix notation could lead to performance problems in some cases.